### DISCUSSION TITLE
Export SQL statements in Table Editor

### DISCUSSION THREAD

**Discussion by nibtime** [supabase/supabase#6683] [Feature Requests]
# Feature request

Add an export function for SQL statements to the UI editor. 

https://supabase.com/docs/guides/database/tables#creating-tables
Basically, so that I could export the "SQL" tab from the "Dashboard" tab

This would be really helpful for quickly copy&pasting to `ley` migrations and also good learning help for a Postgres noob like me.

Bonus points if there even was a SQL data export on top of the CSV export, that would be helpful for quickly seeding data from production into the preview for testing purposes.

Thank you very much for creating Supabase, it's awesome ❤️

**Comment by MildTomato** [comment]
I really love the idea that any kind of view your are looking at in terms of data.. 
so it might be a table view with some filters, or maybe it is a list of users - we could supply the equivalent "SQL" that you could run yourself to get the same data.

Or even better yet, the supabase-js (or other libraries) you can use, to get the same result.

**Reply by nibtime** [reply]
I am currently trying out a workflow with a legacy data migration task where I use the table editor for quick prototyping and exploration and use DataGrip from IntelliJ for the more advanced stuff like the things I was suggesting here.

I even tend to retreat from my own suggestion, as it could interfere with the simplicity of the table editor. This hybrid workflow on data/SQL with DataGrip + Supabase/PostgresSQL supports my style and needs 100% perfectly, so I will just buy the DataGrip license. Great to have that option with Supabase, with Firestore I couldn't just easily use a powerful SQL IDE.

And yes, I always start with data and its shape. Once those are right, the functions and app/site around it kind of follow magically, it's mysterious :)

**Reply by nibtime** [reply]
@MildTomato 

> Or even better yet, the supabase-js (or other libraries) you can use, to get the same result.

How would I achieve the same result with that? For migrations with [ley](https://github.com/lukeed/ley) I need the actual SQL code to implement the `up` and `down` functions in the JS migration files so ley can execute all ups and downs as a single database transaction. As far as I can see `supabase-js` offers a client, which is what I need in app code to query data via API, but I also like to codify the actual data structures and policies in my apps repository as SQL code, to spin up tables matching my app's data structures in fresh Supabase instances, seed data via script, for playing around, running e2e tests and the such.

As DataGrip offers a perfect implementation of what I meant, I think its best to illustrate the features I've been requesting via screenshots:

<details>
<summary>Get SQL code to generate existing tables</summary>

![2022-05-10 17_59_21](https://user-images.githubusercontent.com/52962482/167676515-a50c6107-5888-4a8b-8b91-3b061c71875e.png)

</details>

<details>
<summary>Get SQL code to update columns in existing tables</summary>

![2022-05-10 18_30_31](https://user-images.githubusercontent.com/52962482/167677925-3096b618-87b9-4627-9fec-97a500de3d67.png)

</details>

<details>
<summary>Export data as SQL inserts</summary>

![2022-05-10 18_19_39](https://user-images.githubusercontent.com/52962482/167679264-a29a36f1-6ac8-4d2e-97f5-bcaac9292883.png)


</details>

Hope this can help to clarify whether something like that would make sense to integrate into the table editor of supabase or not.

**Comment by probert94** [comment]
I am currently starting a project with supabase and I am not sure if I should use the table editor or create the queries myself.
As I am not that confortable with SQL anymore, using the table editor is a lot easier for me.
On the other side, the queries are all nicely listed and could even be versioned together with the project (for example on github) and it would be easy to recreate the whole database in a new project etc.
So if it would be possible to create the tables etc. using the visual editor but then generate the needed SQL, we would have the best of both worlds.

**Reply by oli-g-sk** [reply]
I believe what you're saying is possible, and the question is not about recreating the schema / tables, but populating them with rows.

If I'm understanding you correctly, what you're asking for is actually quite easy using the CLI app, and this section of the docs might be helpful: https://supabase.com/docs/guides/cli/local-development
